community/kernel/src/main/java/org/neo4j/kernel/impl/proc/ReflectiveProcedureCompiler.java

```diff
@@ -592,15 +574,27 @@ public ProcedureSignature signature()
         }
 
         @Override
-        public RawIterator<Object[],ProcedureException> apply( Context ctx, Object[] input, ResourceTracker resourceTracker ) throws ProcedureException
+        public RawIterator<Object[],ProcedureException> apply( Context ctx, Object[] input,
+                ResourceTracker resourceTracker ) throws ProcedureException
         {
             // For now, create a new instance of the class for each invocation. In the future, we'd like to keep
             // instances local to
             // at least the executing session, but we don't yet have good interfaces to the kernel to model that with.
             try
             {
-                // verify and possibly map the input
-                input = verifyInput( "Procedure", signature.name(), signature.inputSignature(), input, indexesToMap );
+                List<FieldSignature> inputSignature = signature.inputSignature();
```

Comment:
I'm not sure why removing `verifyInput` helps, as the only signature member access saving is signature.name(). All other accesses happen in both cases.

Reply:
And now we have duplicate code in three places.